### PR TITLE
[UI v2] feat: Adding DocsLink reusable component

### PR DESCRIPTION
**Pull request by devinvillarosa**
1. Create a re-usable `DocsLink` component that can be re-used throughout the app.
2. Replace Docs link in variables page in favor for new `DocsLink` component

Screenshot 2024-12-03 at 10 29 25 AM

https://github.com/user-attachments/assets/59553b4c-37c9-48de-a224-8802bd270c59

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: ui-v2/src/components/ui/docs-link.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { ExternalLinkIcon } from "lucide-react";
+
+import { Button } from "./button";
+
+// Continually add to the list
+type HrefDocs =
+	| "https://docs.prefect.io/latest/guides/variables/"
+	| "https://docs.prefect.io/v3/develop/task-runners";
```
Comment (desertaxle) [inline ui-v2/src/components/ui/docs-link.tsx:8]: What if we gave human-readable names to the links like 'variables-guide' and 'task-runners-guide' and allow those names to be provided to the `DocsLink` component? That way we could get type safety, make reusing links easier, and encapsulate the actual links from parent components.

Comment (devinvillarosa) [reply ui-v2/src/components/ui/docs-link.tsx:8]: sounds good.

I can do a record, like:

```
{ 'variables-guid': '"https://docs.prefect.io/latest/guides/variables'}
```

and we just need to pass in something like:
```
<DocsLink id="variables-guid" />
```

Comment (devinvillarosa) [reply ui-v2/src/components/ui/docs-link.tsx]: Updated on latest push

File: ui-v2/src/components/ui/docs-link.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { ExternalLinkIcon } from "lucide-react";
+
+import { Button } from "./button";
+
+const DOCS_LINKS = {
+	"concurrency-guide": "https://docs.prefect.io/v3/develop/task-runners",
```
Comment (desertaxle) [inline ui-v2/src/components/ui/docs-link.tsx:6]: If this is for the concurrency page, you'll probably want these two instead:
```suggestion
	"global-concurrency-guide": "https://docs.prefect.io/v3/develop/global-concurrency-limits",
	"task-run-concurrency-guide": "https://docs.prefect.io/v3/develop/task-run-limits"
```

**Review by desertaxle — APPROVED** [review]

**Pull request merged**